feat(angular): Add Angular CLI + Webpack Plugin sourcemaps upload guide (#6404)
getsentry/sentry-docs: src/platform-includes/sourcemaps/overview/javascript.angular.mdx

## Uploading Source Maps in an Angular project

> Sentry uses [releases](/product/releases/) to match the correct source maps to your events.
> This page is a guide on how to create releases and upload source maps to Sentry when bundling your Svelte app.

Reviewer: ```suggestion
Sentry uses [releases](/product/releases/) to match the correct source maps to your events.
This page explains how to generate source maps, set releases, and upload source maps to Sentry either manually or automatically when bundling your Angular app.

```

### Generating Source Maps

To generate source maps, you need to add the `sourceMap` option to your `angular.json` build configuration:

```json {filename:angular.json}
{
  // ...
  "projects": {
    "my-app": {
      "architect": {
        "build": {
          "configurations": {
            "production": {
              "sourceMap": {
                "scripts": true
              }
            }
          }
        }
      }
    }
  }
}
```

> ### Uploading Source Maps
> 
> To upload your Angular project's source maps to Sentry, we recommend one of these two options:
> 
> - [**Angular CLI and Sentry Webpack Plugin**](./uploading/angular-webpack/) <br/>
>   Use the Angular CLI, a custom Angular builder and the Sentry Webpack plugin to set releases and upload source maps automatically when running `ng build`.
> - [**Sentry CLI**](./uploading/cli/)<br/>
>   Upload source maps manually using Sentry-CLI.

Reviewer: ```suggestion
### Two Ways to Upload Source Maps

You can either set releases and upload source maps automatically using the Angular CLI, or upload source maps manually using the Sentry-CLI.

- [**Angular CLI and Sentry Webpack Plugin**](./uploading/angular-webpack/) <br/>
  Use the Angular CLI, a custom Angular builder and the Sentry Webpack plugin to set releases and upload source maps automatically when running `ng build`.
- [**Sentry CLI**](./uploading/cli/)<br/>
  Upload source maps manually using Sentry-CLI.
```

> Take a look at this [guide for further options to upload source maps](./uploading/).

Reviewer: ```suggestion
### Additional Guides and Options

While Angular or Sentry CLI works well for out-of-the-box Angular projects, you may want to consider one of the below for other bundlers or more advanced projects and configurations:

- [Webpack](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/webpack/)
Upload your source maps with our Webpack plugin.
- [TypeScript (tsc)](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/typescript/)
Upload your source maps using tsc and Sentry CLI.
- [Rollup](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/rollup/)
Upload your source maps with the Sentry Rollup Plugin.
- [Vite](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/vite/)
Upload your source maps with the Sentry Vite Plugin.
- [esbuild](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/esbuild/)
Upload your source maps with the Sentry esbuild Plugin.
- [UglifyJS](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/uglifyjs/)
Upload your source maps using UglifyJS and Sentry CLI.
- [SystemJS](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/systemjs/)
Upload your source maps using SystemJS and Sentry CLI.
- [Hosting Publicly](https://sentry-docs-git-lms-angular-sourcemaps-upload.sentry.dev/platforms/javascript/guides/angular/sourcemaps/uploading/hosting-publicly/)
Learn about publicly hosting your source maps, including how to address various security concerns.

```

Reviewer: As a global comment, I would recommend changing the content structure and making the parent "Source Maps" pages a more general description of why source maps are so important in Sentry. I'm happy to partner with you to draft this page. 

I would then add all the above content to the "Uploading Source Maps" page. Splitting this information into 2 smaller pages (Source maps & Uploading source maps) creates a fragmented user experience and a lot of unnecessary duplicate content. Happy to set up time to talk more about this.

Reviewer 2: All these other tools won't work with Angular. So I would definitely not include them here.

Author: Leaving the structural change aside (we're gonna talk about this soon), I don't want to add this extensive list here. Angular CLI and Sentry CLI are the two ways we recommend. All other ones either won't work out of the box or should only be of interest for users if their project is really complex and they heavily customized their Angular app and its build process. So for the 95% this would just clutter the page with potentially confusing content IMO.

Reviewer: Ah, good to know. Let's talk about this more when we meet.
